File: src/player.ts
export class Player {

    private mRating: number = 1200;
    private mWins: number = 0;
    private mLosses: number = 0;

    constructor(
        private mId: number,
        private mName: string
    ) {}

    public get id(): number {
        return this.mId;
    }

    public get name(): string {
        return this.mName;
    }

    public get rating(): number {
        return this.mRating;
    }

    public get wins(): number {
        return this.mWins;
    }

    public get losses(): number {
        return this.mLosses;
    }

    public increaseRating(change: number): void {
        this.mRating += change;
    }

    public reduceRating(change: number): void {
        this.mRating -= change;
    }


}

File: kotlin-server/src/main/kotlin/com/example/kotlinserver/Player.kt
package com.example.kotlinserver

class Player (val id: Int, val name: String) {
    var rating = 1200
        private set
    var wins = 0
        private set
    var losses = 0
        private set

    fun increaseRating(change: Int) {
        rating += change
        wins++
    }

    fun reduceRating(change: Int) {
        rating -= change
        losses--
    }
}